XenAdmin/Wizards/ImportWizard/ImportSelectHostPage.cs

```diff
@@ -121,18 +121,35 @@ private get
                             int.TryParse(rasdType.VirtualQuantity.Value.ToString(), out var vCpusCount))
                         {
                             _ovfVCpusCount.Add(vCpusCount);
-                            if (_ovfMaxVCpusCount == null || _ovfMaxVCpusCount < vCpusCount)
+                            if (_ovfMaxVCpusCount < vCpusCount)
                             {
                                 _ovfMaxVCpusCount = vCpusCount;
                             }
                         }
                         // Memory
                         if (rasdType.ResourceType.Value == 4 && long.TryParse(rasdType.VirtualQuantity.Value.ToString(), out var memory))

```

Comment:
The thing is that xapi uses `long` for memory values, so even if we allow `double` when parsing, we still need to check a few lines further down before creating the VM. See the code in method CreateVm() in Import.cs: the datascheme of the OVF API is using `ulong`, but we still convert to `long` for xapi.

Reply:
> The thing is that xapi uses long for memory values, so even if we allow double when parsing, we still need to check a few lines further down before creating the VM. 

Then let's do that? We should show the warning if someone mistakenly imports an OVF with too much memory. Currently if I import an OVF with, - say - 1 EB of memory, the warning won't show at all.

Reply:
In the commit I've just added I've changed `out var memory` to be parsed as `double`, so it makes the calculations further down a bit more straightforward. I've left `_ovfMemory` a `long` though, because this is what will set the VM's memory further down when we CreateVm().